### PR TITLE
1 of 2: Add support for Create Appointment Type

### PR DESCRIPTION
**Pull request by seantcanavan**
- Makefile to automate common tasks
- Add CreateAppointmentType, CreateAppointmentTypeOpts, CreateAppointmentTypeResult

Ran integration tests in a local CopilotIQ repo and everything was :100: 

Apologies I tend to commit quite frequently. Recommend squashing when merging.

EDIT: Also still figuring out this fork model across two repos so unfortunately these changes are already in 'main' on the copilotiq side. My next PR will thus contain all of these changes and more so this PR will be marked 1 of 2.

Going forward I'll make sure I'm not targetting eleanorhealth/go-athenahealth:main from copilotiq/go-athenahealth:main. Instead, I'll target eleanorhealth/go-athenahealth:main from copilotiq/go-athenahealth:feature_XYZ

### CODE REVIEW

File: athenahealth/appointment_types_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"net/http"
+	"os"
+	"testing"
+)
+
+func TestHTTPClient_CreateAppointmentType(t *testing.T) {
+	assert := assert.New(t)
+
+	h := func(w http.ResponseWriter, r *http.Request) {
+		b, _ := os.ReadFile("./resources/CreateAppointmentType.json")
```
Comment (fterrag) [inline athenahealth/appointment_types_test.go:15]: @seantcanavan This test should include asserting that the query params are set correctly.

Comment (seantcanavan) [reply athenahealth/appointment_types_test.go:15]: @fterrag I might be missing something however this endpoint does not use any query string parameters

Comment (dillonstreator) [reply athenahealth/appointment_types_test.go:15]: I believe what Frank meant was asserting the contents of the request body. For reference, checkout the `CreateClaim` test https://github.com/eleanorhealth/go-athenahealth/blob/9313e357f3eb5ff5d3338b11a2f3539951195b6c/athenahealth/claims_test.go#L72-L100

Comment (seantcanavan) [reply athenahealth/appointment_types_test.go:34]: awesome thank you @dillonstreator

**Review by fterrag — APPROVED** [review]

**Pull request merged**